Libs/MRML/DisplayableManager/vtkMRMLModelDisplayableManager.cxx

```diff
@@ -818,13 +818,36 @@ bool vtkMRMLModelDisplayableManager::OnMRMLDisplayableModelNodeModifiedEvent(
     return false;
   }
 
+  int ndnodes = modelNode->GetNumberOfDisplayNodes();
   if (!this->IsModelDisplayable(modelNode))
   {
-    return false;
+    bool requestRender = false;
```

Comment:
The for loop after this looks very similar the the other one below. Could they be merged? If it makes things slower or more complex if we merge them then we should have a comment that explains why it is better to have two loops.

Reply:
Good point. I've integrated the loops so it is much cleaner now.